core/src/main/scala/org/apache/spark/TaskEndReason.scala

```diff
@@ -191,9 +191,12 @@ case object TaskKilled extends TaskFailedReason {
  * Task requested the driver to commit, but was denied.
  */
 @DeveloperApi
-case class TaskCommitDenied(jobID: Int, partitionID: Int, attemptID: Int) extends TaskFailedReason {
+case class TaskCommitDenied(
+    jobID: Int,
+    partitionID: Int,
+    attemptNumber: Int) extends TaskFailedReason {
```

Comment:
I didn't want this patch's backporting to be blocked over binary incompatibility concerns, hence my decision to use the attempt number here. The unique-within-a-SparkContext TaskAttemptIDs are Longs, which would require a change to this interface.

The parameter renaming is potentially source-incompatible. If we think that this is a concern, then I can roll back to the old incorrect name and can add a comment.